atest/testdata/standard_libraries/collections/dictionary.robot

```diff
@@ -232,7 +232,34 @@ Pop From Dictionary With Default
     Should be equal    ${a}    foo
     Should be True   $dict == {'b': 'val2'}
 
+Check invalid dictionary argument errors
+    [Template]    Validate invalid argument error
+    Copy dictionary
+    Dictionary Should Contain Item             string    1    I'm not a dict, I'm string.    a    b
+    Dictionaries Should Be Equal               string    1    I'm not a dict, I'm string.    ${D2}
+    Dictionaries Should Be Equal               string    2    ${D2}    I'm not a dict, I'm string.
+    Dictionary Should Contain Key              string    1    I'm not a dict, I'm string.    a
+    Dictionary Should Contain Sub Dictionary   string    1    I'm not a dict, I'm string.    ${D2}
+    Dictionary Should Contain Sub Dictionary   string    2    ${D2}    I'm not a dict, I'm string.
+    Dictionary Should Contain Value            string    1    I'm not a dict, I'm string.    a
+    Dictionary Should Not Contain Key          string    1    I'm not a dict, I'm string.    a
+    Dictionary Should Not Contain Value        string    1    I'm not a dict, I'm string.    a
+    Get Dictionary Items
+    Get Dictionary Keys
+    Get Dictionary Values
+    Get from dictionary                        string    1    I'm not a dict, I'm string.    a
+    Keep in dictionary                         string    1    I'm not a dict, I'm string.    a
+    Log Dictionary
+    Pop From Dictionary                        string    1    I'm not a dict, I'm string.    a
+    Remove From Dictionary                     string    1    I'm not a dict, I'm string.    a
+    Set To Dictionary                          string    1    I'm not a dict, I'm string.    a    b
+
 *** Keywords ***
+Validate invalid argument error
+    [Arguments]  ${keyword}    ${type}=string    ${position}=1    ${argument}=I'm not a dict, I'm string.    @{kwargs}
```

Comment:
Should be `@{args}`, not `@{kwargs}`. The latter has a different meaning. Possibly could add additional `&{kwargs}` here and pass those forward to the called keyword as well, but I don't think that's needed.

Reply:
Also, this would be an *excellent* place where to use the brand new named-only arguments syntax (#2896). Would require rebasing the branch and can be done also afterwards.